packages/contracts-bedrock/scripts/DeployPeriphery.s.sol

```diff
@@ -63,14 +63,21 @@ contract DeployPeriphery is Deployer {
 
     /// @notice Deploy the ProxyAdmin
     function deployProxyAdmin() public broadcast returns (address addr_) {
-        ProxyAdmin admin = new ProxyAdmin{ salt: keccak256(bytes("ProxyAdmin")) }({
+        bytes32 initCodeHash = keccak256(abi.encodePacked(type(ProxyAdmin).creationCode, abi.encode(msg.sender)));
+        address preComputedAddress = computeCreate2Address(keccak256(bytes("ProxyAdmin")), initCodeHash);
+        if (preComputedAddress.code.length > 0) {
+          console.log("ProxyAdmin already deployed at %s", preComputedAddress);
+          save("ProxyAdmin", address(preComputedAddress));
```

Comment:
In the case where someone is running this on their machine for the first time, but some of these contracts have already been deployed. Would we still want to save here, so that the artifact gets saved to disk, and then in other places where the artifact is referenced it can still be located? (for example, [this function](https://github.com/ethereum-optimism/optimism/blob/adf55b3a60279e4750ab0682cb92d19921bbd92a/packages/contracts-bedrock/scripts/DeployPeriphery.s.sol#L105-L107) requires that the artifact be on disk)